### ISSUE TITLE
SignInRedirect not working behind Load balancer.

### ISSUE THREAD

**Issue by Bryson14** [authts/react-oidc-context#1371]
hello, I'm trying to host a website on AWS. the setup is a vite website, put into a docker container, running on ECS and then getting network through AWS app load balancers. When using this to connect to a company IdP, it works fine in local and when runnning the docker container locally. But when I try to log in when deployed on AWS, the signInRedirect triggers getting the well-known configuration from the IdP, but then doesn't do anything else. I've exhausted a lot of options, so after looking into network configurations on the ALB and ECS, I think it might be the library.

## package.json
```json
{
    "name": "oncoagent-ui",
    "private": true,
    "version": "0.0.0",
    "type": "module",
    "scripts": {
        "start": "vite",
        "build": "tsc -b && vite build",
        "lint": "eslint .",
        "format": "prettier --write .",
        "copyright-check": "node -e \"console.log((new Date()).getFullYear())\" | xargs license-check-and-add check -f copyright-config.json -r",
        "copyright-add": "node -e \"console.log((new Date()).getFullYear())\" | xargs license-check-and-add add -f copyright-config.json -r",
        "preview": "vite preview"
    },
    "dependencies": {
        "@headlessui/react": "^2.1.8",
        "axios": "1.7.3",
        "classnames": "2.5.1",
        "lodash.at": "4.6.0",
        "lucide-react": "^0.441.0",
        "oidc-client-ts": "^3.0.1",
        "react": "18.3.1",
        "react-dom": "18.3.1",
        "react-oidc-context": "^3.1.1",
        "react-router-dom": "6.26.0",
        "react-slider": "2.0.6",
        "react-textarea-autosize": "^8.5.3",
        "zustand": "4.5.4"
    },
    "devDependencies": {
        "@eslint/js": "9.8.0",
        "@svgr/webpack": "6.5.1",
        "@types/lodash.at": "4.6.9",
        "@types/node": "22.3.0",
        "@types/react": "18.3.3",
        "@types/react-dom": "18.3.0",
        "@types/react-slider": "1.3.6",
        "@vitejs/plugin-react": "4.3.1",
        "autoprefixer": "10.4.20",
        "browserslist-to-esbuild": "1.2.0",
        "eslint": "9.8.0",
        "eslint-plugin-react-hooks": "5.1.0-rc.0",
        "eslint-plugin-react-refresh": "0.4.9",
        "globals": "15.9.0",
        "license-check-and-add": "4.0.5",
        "postcss": "8.4.41",
        "prettier": "3.3.3",
        "sass": "1.77.8",
        "tailwindcss": "3.4.9",
        "typescript": "5.5.3",
        "typescript-eslint": "8.0.0",
        "vite": "5.4.0",
        "vite-plugin-svgr": "4.2.0",
        "vite-tsconfig-paths": "4.2.1"
    }
}

```

## App.tsx

```tsx
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Main } from './pages/Main/Main.tsx';
import MainLayout from './pages/layout.tsx';
import { AuthProvider } from 'react-oidc-context';
import { User } from 'oidc-client-ts';

const idpClientId = import.meta.env.VITE_CLIENT_ID;
const idpCallbackUrl = import.meta.env.VITE_CALLBACK_URL;
const idpDomainName = import.meta.env.VITE_DOMAIN;

const authorityEndpoint = `https://company.io/t/${idpDomainName}/oauth2/token/`;

// remove ?code... in params after signin
const onSigninCallback = (_user: User | void): void => {
    window.history.replaceState({}, document.title, window.location.pathname);
};
const oidcConfig = {
    authority: authorityEndpoint,
    client_id: idpClientId,
    redirect_uri: idpCallbackUrl,
    automaticSilentRenew: true,
    onSigninCallback: onSigninCallback
};


export const App = () => {
    return (
        <AuthProvider {...oidcConfig}>
            <BrowserRouter>
                <Routes>
                    <Route path='/' element={<MainLayout />}>
                        <Route index element={<Main />} />
                    </Route>
                </Routes>
            </BrowserRouter>
        </AuthProvider>
    );
};

export default App;
```

## Layout.tsx

```tsx

export default function MainLayout() {
    const auth = useAuth();
    const [hasTriedSignin, setHasTriedSignin] = useState(false);
    const [showButton, setShowButton] = useState(false);
    const handleLogin = () => {
        void auth.signinRedirect();
    };
    const handleLogout = () => void auth.removeUser();

    // Automatic Sign In
    useEffect(() => {
        if (
            !hasAuthParams() &&
            !auth.isAuthenticated &&
            !auth.activeNavigator &&
            !auth.isLoading &&
            !hasTriedSignin
        ) {
            auth.signinRedirect();
            setHasTriedSignin(true);
        }
    }, [auth, hasTriedSignin]);

    useEffect(() => {
        const timer = setTimeout(() => {
            setShowButton(true);
        }, 2000);

        return () => clearTimeout(timer);
    }, []);

    if (!auth.isAuthenticated || !auth.user) {
     <LoginPageWithButton />
     }
     
     return <Header><Outlet /></Header>
```

So this is what happens in local:

I start the site, and I either automatically get signed in, (or if there is some bugginest, i have to click the button. Sometimes a couple times: i see this picture and on the network tab I can see the various calls to the company IdP then I get redirected to the company SSO page:

![image](https://github.com/user-attachments/assets/29bf705e-b8ee-4c2f-a2d6-d29bcb42c021)

But on AWS I just get nothing after the initial well-known network request:

![image](https://github.com/user-attachments/assets/35397675-a854-4f1b-abea-ec76894f7a08)

Here is the network request:

![image](https://github.com/user-attachments/assets/4677dff9-906d-4cc9-a50a-1c43399737ed)

So I'm pretty stuck at this point. Not sure why things work sometimes and not behind a load balancer. But even then,. i'm not uysing SSR, so if i have the js bundles, my browser should make the request to the IdP.

**Comment by Bryson14** [comment]
On the deployed site, clikcing login doesn't nothing else: The code to the button just looks like this:

```tsx
{showButton ? (
                        <button
                            onClick={() => {
                                console.log('logging in');
                                auth.signinRedirect();
                            }}
                            className='bg-slate-200 text-black px-4 py-2 rounded-md hover:text-white hover:bg-purple-500 focus:text-white focus:bg-purple-500 transition-colors focus:outline-none focus:ring-2 focus:ring-purple-500 focus:ring-offset-2'
                        >
                            {translate('header.login')}
                        </button>
                    ) : (
                        <Loader2 className='w-8 h-8 animate-spin text-blue-500 mx-auto' />
                    )}
                    ```

**Comment by Bryson14** [comment]
Solved: The issue that my load balancer was not using HTTPS. By adding a cert and upgrading the connection to the deployed site from HTTP to HTTPS, the auth library would log in. I would have liked to known which event I could have used  in order see that error in the console, but I discovered it through experimentation. 

I was not using the client secret, but using PKCE. Also, I got the same "no response" error when I would temporarily allow for HTTP connections on my firefox browser.
